pkg/issuer/acme/http/pod.go

```diff
@@ -168,6 +169,9 @@ func (s *Solver) buildDefaultPod(ch *cmacme.Challenge) *corev1.Pod {
 		},
 		Spec: corev1.PodSpec{
 			RestartPolicy: corev1.RestartPolicyOnFailure,
+			SecurityContext: &corev1.PodSecurityContext{
+				RunAsNonRoot: pointer.BoolPtr(true),
+			},
```

Comment:
This fixes: https://github.com/jetstack/cert-manager/issues/3721, but note that Kyverno did not enforce this, probably due to a bug in Kyverno  where it ignores Pods that are created with an OwnerReference. See https://github.com/kyverno/kyverno/issues/1784#issuecomment-845915946